paasta_tools/check_services_replication_tools.py

```diff
@@ -0,0 +1,122 @@
+# Copyright 2015-2019 Yelp Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import argparse
+import logging
+from typing import Callable
+from typing import Sequence
+from typing import Type
+
+from mypy_extensions import Arg
+
+from paasta_tools.kubernetes_tools import get_all_nodes
+from paasta_tools.kubernetes_tools import get_all_pods
+from paasta_tools.kubernetes_tools import KubeClient
+from paasta_tools.kubernetes_tools import V1Pod
+from paasta_tools.paasta_service_config_loader import PaastaServiceConfigLoader
+from paasta_tools.smartstack_tools import KubeSmartstackReplicationChecker
+from paasta_tools.utils import DEFAULT_SOA_DIR
+from paasta_tools.utils import InstanceConfig_T
+from paasta_tools.utils import list_services
+from paasta_tools.utils import load_system_paasta_config
+
+
+log = logging.getLogger(__name__)
+
+
+def parse_args() -> argparse.Namespace:
```

Comment:
can you add an arg for `service.instance` and then default to checking all if it isn't provided? that'll make this easier to do the `list_kubernetes_service_instances|shuf|xargs` trick in the future if this starts taking too long

Reply:
done